numpy/lib/function_base.py

```diff
@@ -2699,6 +2717,14 @@ def __init__(self, pyfunc, otypes=None, doc=None, excluded=None,
         else:
             self._in_and_out_core_dims = None
 
+    def __get__(self, instance, owner):
+        return type(self)(self.pyfunc.__get__(instance, owner),
+                          otypes=self.otypes,
+                          doc=self.__doc__,
+                          excluded=self.excluded,
+                          cache=self.cache,
+                          signature=self.signature)
```

Comment:
Does this work correctly for `MyClass.vec_method(myinstance, ...)`? If not, then does it work if `self` is in `excluded`? In that case, does a direct method call still work?